airflow/providers/cohere/hooks/cohere.py

```diff
@@ -46,23 +59,34 @@ def __init__(
         conn_id: str = default_conn_name,
         timeout: int | None = None,
         max_retries: int | None = None,
+        request_options: dict | None = None,
     ) -> None:
         super().__init__()
         self.conn_id = conn_id
         self.timeout = timeout
         self.max_retries = max_retries
+        self.request_options = request_options
+        if self.max_retries:
+            warnings.warn(
+                "Argument `max_retries` is deprecated. Please use `request_options` dict for function-specific request configuration instead.",
+                AirflowProviderDeprecationWarning,
+                stacklevel=2,
+            )
+            self.request_options = (
+                {"max_retries": self.max_retries}
+                if self.request_options is None
+                else self.request_options.update({"max_retries": self.max_retries})
```

Comment:
Looks like `self.request_options` will be `None` everytime in `else` case. 

![Screenshot 2024-05-21 at 2 47 44 PM](https://github.com/apache/airflow/assets/13021213/30beb5fd-1f68-4b98-beed-b0c8dfe31f6b)